src/components/RegistrationForm.tsx

```diff
@@ -21,81 +26,260 @@ const labelStyle = {
-                        Confirm Password
-                    </FormLabel>
-                    <Input
-                        id="confirmPassword"
-                        name="confirmPassword"
-                        type="password"
-                        style={inputStyle}
-                    />
-                </FormControl>
-                <RadioGroup id="role" name="role" width="22em">
-                    <Stack direction="row" display="flex" alignItems="center">
-                        <Radio value="parent" flex="1" bg="white" size="md">
-                            <span style={radioLabelStyle}>Parent</span>
-                        </Radio>
-                        <Radio value="tutor" flex="1" bg="white" size="md">
-                            <span style={radioLabelStyle}>Tutor</span>
-                        </Radio>
+        <Formik
+            initialValues={initialValues}
+            validationSchema={Yup.object({
+                firstName: Yup.string()
+                    .required('Please provide your first name')
+                    .min(2, 'First name must be at least 2 characters')
+                    .max(20, 'First name should not be more than 20 characters'),
+
+                lastName: Yup.string()
+                    .required('Please provide your last name')
+                    .max(20, 'Last name should not be more than 20 characters'),
+                email: Yup.string()
+                    .email('Please provide a valid email address')
+                    .required('Please provide your email'),
+                password: Yup.string()
+                    .min(6, 'Password must be at least 6 characters')
+                    .required('Please provide password')
+                    .matches(
+                        /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[\W_]).{6,}$/,
+                        'Password must contain at least one lowercase letter, uppercase letter, number, and special character'
+                    ),
+                confirmPassword: Yup.string()
+                    .oneOf([Yup.ref('password'), null], 'Passwords must match')
+                    .required('Confirmation of password is required'),
+                role: Yup.string().required('Please select a role'),
+            })}
+            onSubmit={async (values, { setSubmitting, setStatus, setFieldError }) => {
+                setStatus('Processing');
+                try {
+                    const requestData = {
+                        firstName: values.firstName,
+                        lastName: values.lastName,
+                        email: values.email,
+                        password: values.password,
+                        role: values.role,
+                    };
+                    //console.log('Request Data:', requestData);
+                    const response = await axios.post(
+                        'https://ffprac-team2-back.onrender.com/api/v1/auth/register',
```

Comment:
For best practices I would recommend to hide the full URL used for making requests in the .env file. This guarantees a more secure management of confidential data such as API URLs and helps the process of changing URLs without modifying the application code. I would use something like REACT_APP_BASE_URL for instance.

Reply:
Done